src/v0/destinations/pardot/networkHandler.js

```diff
@@ -46,6 +46,20 @@ const getStatus = (code) => {
 const pardotRespHandler = (destResponse, stageMsg) => {
   const { status, response } = destResponse;
   const respAttributes = response['@attributes'];
+
+  // to handle errors like service unavilable, wrong url, no response
+  if (!respAttributes) {
+    throw new NetworkError(
+      `${JSON.stringify(response)} ${stageMsg}`,
+      status,
+      {
+        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(status),
+      },
+      response,
+      getAuthErrCategory(status),
```

Comment:
We are passing `http` Status but we are expecting `code` in `getAuthErrCategory` function